api/src/Appointments/State/ExportAppointmentCollectionProvider.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Appointments\State;
+
+use ApiPlatform\Metadata\Operation;
+use ApiPlatform\State\ProviderInterface;
+use App\Appointments\Resource\AppointmentCsv;
+use App\Entity\Appointment;
+use App\Repository\AppointmentRepository;
+
+/**
+ * @template-implements ProviderInterface<AppointmentCsv>
+ */
+final readonly class ExportAppointmentCollectionProvider implements ProviderInterface
+{
+    public function __construct(
+        private AppointmentRepository $repository,
+    ) {
+    }
+
+    /**
+     * @return AppointmentCsv[]
+     */
+    public function provide(Operation $operation, array $uriVariables = [], array $context = []): array
+    {
+        /** @var array<array-key, Appointment> $collection */
+        $collection = $this->repository->findBy([], ['id' => 'DESC']);
+
+        return array_map(static function (Appointment $appointment) {
+            return new AppointmentCsv(
+                $appointment->customer ? $appointment->customer->lastName : '',
+                $appointment->customer ? $appointment->customer->firstName : '',
```

Comment:
Yes, I tried to do that but sometimes the `customerName` field seems to be null even if a customer is set, so I prefer to do this way to avoid checking if all these properties are empty: `$appointment->customer`, $appointment->customerName then if it null lastName and firstName